storage/src/persistent/database.rs

```diff
@@ -29,6 +32,20 @@ pub enum DBError {
     MissingColumnFamily { name: &'static str },
     #[fail(display = "Database incompatibility {}", name)]
     DatabaseIncompatibility { name: String },
+    #[fail(display = "Value already exists {}", key)]
+    ValueExists { key: String },
+    #[fail(display = "Guard Poison {} ", error)]
+    GuardPoison { error: String },
```

Comment:
@mat13mn 
what is difference between GuardPoison  vs LockError?
I think it is the same, could we just have just LockError?